test/expint.jl

```diff
@@ -158,6 +158,11 @@ using Base.MathConstants
                 Γ, cf, iter = SpecialFunctions.En_cf_gamma(1 / 2, x)
                 @test Γ + cf*exp(-x) ≈ y
             end
+            # type stability
+            @inferred SpecialFunctions.En_cf_gamma(1, 1.0 + 2.1im, 1000)
+            @test SpecialFunctions.En_cf_gamma(1, 1.0 + 2.1im, 1000) isa Tuple{ComplexF64,ComplexF64,Int}
+            @inferred SpecialFunctions.En_cf_gamma(1, 1.0f0, 1000)
+            @test SpecialFunctions.En_cf_gamma(1, 1.0f0, 1000) isa Tuple{Float32,Float32,Int}
```

Comment:
Minor improvement: You can combine these tests to
```suggestion
            @test @inferred(SpecialFunctions.En_cf_gamma(1, 1.0 + 2.1im, 1000)) isa Tuple{ComplexF64,ComplexF64,Int}
            @test @inferred(SpecialFunctions.En_cf_gamma(1, 1.0f0, 1000)) isa Tuple{Float32,Float32,Int}
```

Reply:
I thinks this would not work as `@inferred` only check the return type of `isa` exprission
 
```julia
julia> f(a) = a > 1 ? 1 : 1.0
f (generic function with 1 method)

julia> @test @inferred f(2) isa Int
Test Passed

julia> @inferred f(2)
ERROR: return type Int64 does not match inferred return type Union{Float64, Int64}
Stacktrace:
 [1] error(s::String)
   @ Base .\error.jl:35
 [2] top-level scope
   @ REPL[11]:1

```

Reply:
sorry, I miss the `@inferred(..)` here